numpy/core/function_base.py

```diff
@@ -1,10 +1,24 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
+def _index_deprecate(i, stacklevel=2):
```

Comment:
Note that using index passes numpy booleans, although python booleans always pass as subclasses of int. Also, we are counting (integers) not indexing. If we check for indexes we should change the documentation and warning to reflect that requirement.